### PR TITLE
[FIX] sale: context lang used on product variants display name

### PR DESCRIPTION
**Pull request by jhk-odoo**
Steps to reproduce:
- install sales and contact
- install another language
- go to contacts, pick a specific contact and set its language to
the new language installed
- go to sales > settings > activate the "product configurator" option
- create a product with a translated variant (the variant has to have
at least 2 translated attributes with custom values)
- create a quotation with the modified contact and the product
with the variant (use the product configurator wizard) > save
- print that quotation

Previous behavior:
product description is not fully translated
the part describing the variants is in the current user's language
while the product display_name is using the associated partner's
language

Current behavior:
product description is fully translated into the correct language

opw-2168102

### CODE REVIEW

**Comment by jhk-odoo** [comment]
@jpp-odoo 
@nle-odoo

File: addons/sale/models/sale.py [hunk 1]
```diff
@@ -1619,16 +1619,21 @@ def _get_sale_order_line_multiline_description_variants(self):
 
         name = "\n"
 
+        #variants need to be displayed in the language of the receiving user
+        wanted_lang = self._context.get('lang') or 'en_US'
```
Comment (jpp-odoo) [inline addons/sale/models/sale.py]: You should use get_lang : https://github.com/odoo/odoo/blob/13.0/odoo/tools/misc.py#L1167
```suggestion
        wanted_lang = get_lang(self.env)
```

**Comment by nim-odoo** [comment]
ping @jpp-odoo

**Comment by jhk-odoo** [comment]
updated based on @nle-odoo comment

File: addons/sale/models/sale.py [hunk 2]
```diff
@@ -1619,16 +1619,21 @@ def _get_sale_order_line_multiline_description_variants(self):
...
+            wanted_lang = self.order_id.partner_id.lang
+
         custom_ptavs = self.product_custom_attribute_value_ids.custom_product_template_attribute_value_id
         no_variant_ptavs = self.product_no_variant_attribute_value_ids._origin
 
         # display the no_variant attributes, except those that are also
         # displayed by a custom (avoid duplicate description)
         for ptav in (no_variant_ptavs - custom_ptavs):
-            name += "\n" + ptav.display_name
+            name += "\n" + ptav.with_context(lang=wanted_lang).display_name
```
Comment (jpp-odoo) [inline addons/sale/models/sale.py]:
```suggestion
            name += "\n" + ptav.with_context(lang=self.order_id.partner_id.lang).display_name
```

File: addons/sale/models/sale.py [hunk 3]
```diff
@@ -1619,16 +1619,21 @@ def _get_sale_order_line_multiline_description_variants(self):
...
 
         # display the is_custom values
         for pacv in self.product_custom_attribute_value_ids:
-            name += "\n" + pacv.display_name
+            name += "\n" + pacv.with_context(lang=wanted_lang).display_name
```
Comment (jpp-odoo) [inline addons/sale/models/sale.py]:
```suggestion
            name += "\n" + pacv.with_context(lang=self.order_id.partner_id.lang).display_name
```

**Comment by nle-odoo** [comment]
robodoo r+

**Pull request closed without merging**